### PR TITLE
Catch and reraise dates errors

### PR DESCRIPTION
**Pull request by esoergel**
https://dimagi-dev.atlassian.net/browse/HI-567
Apparently xls allows nonsensical or ambiguous dates.  The xlrd [docs](https://xlrd.readthedocs.io/en/latest/dates.html) put it well:

> **Dates in Excel spreadsheets**
> In reality, there are no such things. What you have are floating point numbers and pious hope.

xlrd detects these nonsensical values and raises an error.  Catching them like this should make it a little easier to see what's going on.

I also looked in to catching this error in the case import code and presenting it to the user as an error, but ended up punting on that for a few reasons:

* This is very rare
* If I switch from a for loop to a while loop with `next()` calls inside a try/except, the iterator will stop working once the exception is hit, so the remaining rows won't be processed
* Returning anything other than a usable value changes this API, and all usages would need to account for that.  This seems like more than I want to bite off.

Other options I considered were:

* Just return `None`, like we do for an empty cell.
* Force the value to be a date - this is what other spreadsheet programs do, and what I'm tempted by if this issue recurs.  For instance, `xlrd.xldate_as_datetime` does manage to spit out a datetime when confronted by this particular issue, but it's not meant to be used like that (it doesn't support dateless times)

In short, this is where I've landed, but I'm open to second opinions.

### CODE REVIEW

File: corehq/util/workbook_reading/adapters/xls.py [hunk 1]
```diff
@@ -1,14 +1,22 @@
-from __future__ import absolute_import
-from __future__ import unicode_literals
+from __future__ import absolute_import, unicode_literals
```
Comment (kaapstorm) [inline corehq/util/workbook_reading/adapters/xls.py]: I keep undoing this particular change when isort does this to my imports, because I vaguely remember @nickpell saying that it was useful to have these on separate lines, for scripting or grepping or something. I was hoping we could get isort to "force_single_line" just for `__future__` imports, and use "multi_line_output = 3" for everything else, but I don't think that's an option.

If this change is fine then I'll stop undoing it in my own code, and if it's not, consider this comment a flag.

File: corehq/util/workbook_reading/adapters/xls.py [hunk 2]
```diff
@@ -38,7 +46,12 @@ def _make_cell_value(self, cell):
             else:
                 return cell.value
         elif cell.ctype == xlrd.XL_CELL_DATE:
-            datetime_tuple = xlrd.xldate_as_tuple(cell.value, self._sheet.book.datemode)
+            try:
+                datetime_tuple = xlrd.xldate_as_tuple(cell.value, self._sheet.book.datemode)
+            except xlrd.XLDateError as e:
+                # https://xlrd.readthedocs.io/en/latest/dates.html
```
Comment (kaapstorm) [inline corehq/util/workbook_reading/adapters/xls.py]: (This is horrifying.)

**Review by kaapstorm — APPROVED** [review]

File: corehq/util/workbook_reading/adapters/xls.py (hunk 1 above)
Comment (nickpell) [reply corehq/util/workbook_reading/adapters/xls.py]: I think the scripting only cares about having ```print_function``` on a separate line, so this change is fine.
